table/block.cc

```diff
@@ -282,14 +293,56 @@ bool BlockIter::ParseNextKey() {
     }
 
     value_ = Slice(p + non_shared, value_length);
-    while (restart_index_ + 1 < num_restarts_ &&
-           GetRestartPoint(restart_index_ + 1) < current_) {
+    uint32_t next_restart_point = 0;
+    bool next_inited = false;
+    while (restart_index_ + 1 < num_restarts_ && (next_inited = true) &&
+           (next_restart_point = GetRestartPoint(restart_index_ + 1)) <
+               current_) {
       ++restart_index_;
     }
+    if (value_delta_encoded_) {
+      assert(value_length == 0);
+      DecodeCurrentValue(next_inited, next_restart_point, p, non_shared);
+    }
     return true;
   }
 }
 
+// The fomrat:
+// restart_point   0: k, v (offset, size), k, v (size), ..., k, v (size)
```

Comment:
Do we consider:
```
k, v (offset, size1), k, v (size2-size1), ..., k, v (size3-size2)
```
as what @ot used to proposed?

I also hope
```
k, v (offset, size1), k, v [padding_size](size2-size1), ..., k, v [padding_size](size3-size2)
```

Reply:
I think the main factor of whether to take this approach is that whether it is common to have block size difference to within 64 bytes. Sounds like very unlikely though. So maybe size is good.

Reply:
If you're referring to https://www.prod.facebook.com/groups/rocksdb.dev/permalink/815533521878497/, I did not suggest that, I was just proposing to use offset, size, size, size, ... like here, and if we have no other ways to distinguish restart points from delta handles, use negative size (with zigzag encoding) to use the sign bit to signal that.

Other than that, I also still believe that we should try to drop the payload size byte here (is it done in this PR?) and switch to sub-varint for prefix coding.